persist/sqlite/consensus_refactored_test.go

```diff
@@ -0,0 +1,275 @@
+package sqlite_test
+
+import (
+	"math"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"go.sia.tech/core/consensus"
+	"go.sia.tech/core/types"
+	"go.sia.tech/coreutils/chain"
+	ctestutil "go.sia.tech/coreutils/testutil"
+	"go.sia.tech/explored/explorer"
+	"go.sia.tech/explored/internal/testutil"
+	"go.sia.tech/explored/persist/sqlite"
+	"go.uber.org/zap/zaptest"
+)
+
+type node struct {
+	db    explorer.Store
+	store *chain.DBStore
+
+	blocks []types.Block
+	states []consensus.State
+}
+
+func newStoreRefactored(t *testing.T, v2 bool, f func(*consensus.Network, types.Block)) *node {
+	log := zaptest.NewLogger(t)
+	dir := t.TempDir()
+
+	db, err := sqlite.OpenDatabase(filepath.Join(dir, "explored.sqlite3"), log.Named("sqlite3"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var network *consensus.Network
+	var genesisBlock types.Block
+	if v2 {
+		network, genesisBlock = ctestutil.V2Network()
+	} else {
+		network, genesisBlock = ctestutil.Network()
+	}
+	if f != nil {
+		f(network, genesisBlock)
+	}
+
+	store, genesisState, err := chain.NewDBStore(chain.NewMemDB(), network, genesisBlock, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bs := consensus.V1BlockSupplement{Transactions: make([]consensus.V1TransactionSupplement, len(genesisBlock.Transactions))}
+	_, au := consensus.ApplyBlock(network.GenesisState(), genesisBlock, bs, time.Time{})
+	if err := db.UpdateChainState(nil, []chain.ApplyUpdate{{
+		ApplyUpdate: au,
+		Block:       genesisBlock,
+		State:       genesisState,
+	}}); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Cleanup(func() {
```

Comment:
`nit` call `t.Cleanup(func() {db.Close()})` immediately after opening the database